src/coreclr/src/jit/optcse.cpp

```diff
@@ -498,8 +498,22 @@ unsigned Compiler::optValnumCSE_Index(GenTree* tree, Statement* stmt)
 
                 /* Start the list with the first CSE candidate recorded */
 
-                hashDsc->csdTreeList = newElem;
-                hashDsc->csdTreeLast = newElem;
+                hashDsc->csdTreeList  = newElem;
+                hashDsc->csdTreeLast  = newElem;
+                hashDsc->csdStructHnd = NO_CLASS_HANDLE;
+
+                hashDsc->csdStructHndMismatch = false;
+
+                if (varTypeIsStruct(tree->gtType))
+                {
+                    // When we have a GT_IND node with a SIMD type then we don't have a reliable
+                    // struct handle and gtGetStructHandleIfPresent returns a guess that can be wrong
+                    //
```

Comment:
This is a workaround for the case where we have GT_IND (of simd type) and they don't have a class handle.
We now record  the Struct Handle at each CSE location and set csdStructHndMismatch to true if any of them are mismatched.  We also use the csdStructHnd to create the CSE LclVar when performing a CSE replacement.

This code can be removed when gtGetStructHandleIfPresent stops guessing and GT_IND nodes always have valid struct handles.

Reply:
It should probably have a `TODO` comment.

Reply:
added